src/instance.ts

```diff
@@ -365,22 +365,31 @@ class Instance extends common.GrpcServiceObject {
    * @property {number} [pageSize] Maximum number of results per page.
    * @property {string} [pageToken] A previously-returned page token
    *     representing part of the larger set of results to view.
+   * @property {object} [gaxOptions] Request configuration options, outlined
```

Comment:
How does it work that we have top-level "pageSize" and "pageToken", but they're also in `gaxOptions` (e.g. "gaxOptions.pageSize")?

Reply:
Thank you!

The existing behavior is as follows:

Current options for listing methods typically look something like this
```ts
{
	parent?: string;
	pageSize?: number;
	pageToken?: string;
	filter?: string;
	autoPaginate?: boolean;
	maxApiCalls?: number;
	gaxOptions?: CallOptions;
}
```
Where `gaxOptions` (among others) contain the following options as well
```ts
{
	autoPaginate?: boolean;
    	pageToken?: string;
    	pageSize?: number;
}
```
Which is utilized as 
```ts
instance.getBackups({
        pageSize: 3,
        pageToken,
        gaxOptions: {autoPaginate: false},
});
```
in [system tests](https://github.com/googleapis/nodejs-spanner/blob/5624b7bafb585adcbd0a7c9d53f728ff77afb1fa/system-test/spanner.ts#L1136-L1140) as well as [samples](https://github.com/googleapis/nodejs-spanner/blob/5624b7bafb585adcbd0a7c9d53f728ff77afb1fa/samples/backups-get.js#L100-L104).


In terms of `reqOpts` and `gaxOpts` of the `request()` method the following options are respected
```ts
reqOpts = {
	parent: string;
	pageSize?: number;
	pageToken?: string;
	filter?: string;
}
gaxOpts = {
	autoPaginate?: boolean;
}
```
having `pageSize` and `pageToken` in two places is confusing and misleading. Should we just remove them from the top-level and pass to the `reqOts` from the the `gaxOptions` object, or accept and respect `pageSize` and `pageToken` values from both top-level and `gaxOptios`, or another option?
WDYT @stephenplusplus

Reply:
This is indeed tricky! I don't think we have the option to remove it from the top-level, since that would conflict with the raw API documentation, e.g. https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.database.v1#google.spanner.admin.database.v1.ListBackupsRequest which accepts "pageSize" and "pageToken".

Do you know where & what gax is doing with "pageSize" and "pageToken". It seems like the wrong place to send a native request parameter, as opposed to the other options in "gaxOptions" which are all custom helpers like "autoPaginate", "bundleOptions", etc.

Reply:
> Do you know where & what gax is doing with "pageSize" and "pageToken". It seems like the wrong place to send a native request parameter, as opposed to the other options in "gaxOptions" which are all custom helpers like "autoPaginate", "bundleOptions", etc.

Perhaps these values where used by [`paginator`](https://github.com/googleapis/nodejs-paginator/blob/cabc0489190cfae73178354841286a3e89779d3f/src/index.ts#L41-L53).

Speaking of which, I believe it (`paginator.streamify`) could be dropped as both [paginated](https://github.com/googleapis/nodejs-spanner/blob/5624b7bafb585adcbd0a7c9d53f728ff77afb1fa/src/v1/database_admin_client.ts#L2039) and [streaming](https://github.com/googleapis/nodejs-spanner/blob/5624b7bafb585adcbd0a7c9d53f728ff77afb1fa/src/v1/database_admin_client.ts#L2150) calls can be handled now.

Reply:
Should gax stop listing those under CallOptions?

Reply:
refactored to pass `pageSize` and `pageToken` as `reqOpts`.
PTAL

Reply:
Thanks for trying to clean up this mess @AVaksman. I'm not a fan of having to support `pageSize` and `pageToken` from `CallOptions` but I feel like we have no choice :( Given that it's in the documentation, it's very likely someone will try to set it in `gaxOptions` and wonder why it's not being respected.